app/models/harvest_definition.rb

```diff
@@ -40,4 +40,8 @@ def to_h
       }
     }
   end
+
+  def clone(pipeline)
+    HarvestDefinition.new(dup.attributes.merge(pipeline:))
```

Comment:
Maybe dumb question, but couldn't work out where `dup` came from. What is it?

Reply:
Also not sure what `dup.attributes.merge(pipeline:)` does.  Just letting you know in case you expect it to be obvious, so you can decide if it needs a comment.

Reply:
Not a dumb question! `dup` is an ActiveRecord method which returns a copy of the record but it doesn't have the id or the created_at. It also doesn't bring across the associations. 

`dup.attributes` returns a hash of new attributes that I want to assign to the `HarvestDefinition` and then I merge the `pipeline` argument into it to assign the `HarvestDefinition` to the new pipeline, rather than the old one that was in the duplicates attributes. 

https://apidock.com/rails/ActiveRecord/Core/dup